plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1463,7 +1463,7 @@ public Answer createVolume(final CreateObjectCommand cmd) {
             primaryPool = storagePoolMgr.getStoragePool(primaryStore.getPoolType(), primaryStore.getUuid());
             disksize = volume.getSize();
             PhysicalDiskFormat format;
-            if (volume.getFormat() == null) {
+            if (volume.getFormat() == null || StoragePoolType.RBD.equals(primaryStore.getPoolType())) {
```

Comment:
@slavkap this might be applicable to volumes on other storage pools as well, as the format depends on the underlying pool. I think, it is better to set the right volume format based on the primary storage in the volume orchestrator, instead of overriding here based on the pool type when the format is null. (or) update the correct format in the the primary storage driver while creating volume.

Reply:
Hi @sureshanaparti, thank you for the review! I wondered where to set the correct format. When we attach an allocated  Ceph volume, we set RAW format here. That's why I decided to do this in one place and in my case.

Reply:
To set the format after volume creation, check here: https://github.com/apache/cloudstack/blob/b32f3697e6f98ca209bba222fa8c58112f94d85f/engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeObject.java#L663

Reply:
@slavkap @sureshanaparti do we need more changes, or is this PR ready for testing?

Reply:
> To set the format after volume creation, check here:
> 
> https://github.com/apache/cloudstack/blob/b32f3697e6f98ca209bba222fa8c58112f94d85f/engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeObject.java#L663
> 
> Irrespective of the current disk format, update to RAW when pool type is RBD.

Hi @slavkap Can you check if the format can be updated here. Thanks.

Reply:
Hi @sureshanaparti, I can check, but the correct volume format is set here if you create an allocated RBD volume. Therefore, in my case also, I prefer the format to be set here.